File: backend/Watchdog.Core/Watchdog.Core.DAL/Entities/Team.cs
using System.Collections.Generic;
using Watchdog.Core.DAL.Entities.Common;

namespace Watchdog.Core.DAL.Entities
{
    public class Team : AuditEntity<int>
    {
        public Team()
        {
            Members = new List<Member>();
            ApplicationTeams = new List<ApplicationTeam>();
        }

        public string Name { get; set; }

        public User User { get; set; }

        public int OrganizationId { get; set; }
        
        public Organization Organization { get; set; }

        public ICollection<Member> Members { get; set; }

        public ICollection<ApplicationTeam> ApplicationTeams { get; set; }
    }
}

File: frontend/src/app/shared/models/team/team.ts
import { User } from '@core/models/user';
import { Organization } from '@shared/models/organization/organization';
import { Member } from '@shared/models/member/member';

export interface Team {
    id: number,
    name: string,
    createdBy: number,
    user: User,
    organizationId: number,
    organization: Organization,
    members: Member[],
    createdAd: Date
}
